Makefile

```diff
@@ -0,0 +1,55 @@
+COLOUR_GREEN=\033[0;32m
+COLOUR_RED=\033[0;31m
+COLOUR_BLUE=\033[0;34m
+COLOUR_END=\033[0m
+
+.DEFAULT_GOAL:=help
+
+SHELL := /bin/bash
+GENERAL_LIB_LOCATION := $(shell pip show rados | grep -oP "(?<=Location: ).*")
+LIBRADOS_LOCATION := $(shell find "${GENERAL_LIB_LOCATION}" -name 'rados*.so' -print)
+LIBRADOS_EGGINFO_LOCATION := $(shell find "${GENERAL_LIB_LOCATION}" -name 'rados*.egg-info' -print)
+LIBRADOS_NAME := $(shell basename "${LIBRADOS_LOCATION}")
+LIBRADOS_EGGINFO_NAME := $(shell basename "${LIBRADOS_EGGINFO_LOCATION}")
+
+## checking if docker, or podman should be used. Podman is preferred.
+ifeq ($(shell command -v podman 2> /dev/null),)
+	CONTAINERCMD=docker
+else
+	CONTAINERCMD=podman
+endif
+
+
+.PHONY: help
+help: ## Display this help message
+	@echo -e '${COLOUR_RED}Usage: make <command>${COLOUR_END}'
+	@cat $(MAKEFILE_LIST) | grep '^[a-zA-Z]'  | \
+	    awk -F ':.*?## ' 'NF==2 {printf "  %-26s%s\n\n", $$1, "${COLOUR_GREEN}"$$2"${COLOUR_END}"}'
+
+.PHONY: setup
+setup: ## Setting up the docker-compose environment
+	${MAKE} setup-venv
+	
+setup-venv:
+	python -m venv ./.venv
+	source ./.venv/bin/activate
+	@echo -e 'GENERAL_LIB_LOCATION: ${GENERAL_LIB_LOCATION}'
+	@echo -e 'LIBRADOS_LOCATION: ${LIBRADOS_LOCATION}'
+	@echo -e 'LIBRADOS_EGGINFO_LOCATION: ${LIBRADOS_EGGINFO_LOCATION}' && \
+	ln -s "${LIBRADOS_LOCATION}" ./.venv/lib/python3.12/site-packages/"${LIBRADOS_NAME}" && \
+	ln -s "${LIBRADOS_EGGINFO_LOCATION}" ./.venv/lib/python3.12/site-packages/"${LIBRADOS_EGGINFO_NAME}" && \
+	pip install -r requirements.txt
+
+.PHONY: build-container
+TAG ?= 0.0.1
```

Comment:
I would suggest to name the variable like in the `Dockerfile`. `TAG` seems to be a bit to common (and too specific to git) for me. In fact it's not the Docker tag or related to it anyway.